packages/shared/src/hooks/squads/usePostToSquad.ts

```diff
@@ -141,25 +160,34 @@ export const usePostToSquad = ({
     },
   });
 
-  const isPosting =
-    isPostLoading || isLinkLoading || isPostSuccess || isLinkSuccess;
+  const isPosting = isPostLoading || isLinkLoading || isPostModerationLoading;
 
-  const isUpdating = isUpdatePostSuccess || isUpdatePostLoading;
+  const isUpdating =
+    isUpdatePostSuccess || isUpdatePostLoading || isPostModerationLoading;
+
+  const isSuccess = didCreatePostModeration || isPostSuccess || isLinkSuccess;
 
   const onSubmitPost = useCallback<UsePostToSquad['onSubmitPost']>(
-    (e, sourceId, commentary) => {
+    (e, squad, commentary) => {
       e?.preventDefault();
-
       if (isPosting) {
         return null;
       }
 
       if (preview.id) {
-        return onPost({
-          id: preview.id,
-          sourceId,
-          commentary,
-        });
+        return moderationRequired(squad)
+          ? onCreatePostModeration({
+              type: PostType.Share,
+              sourceId: squad.id,
+              sharedPostId: preview.id,
+              externalLink: preview?.url,
+              commentary,
```

Comment:
This format is for submitting external link, the parameters should not need the `externalLink`.